include/seqan3/alignment/pairwise_alignment_edit_distance_unbanded.hpp

```diff
@@ -0,0 +1,548 @@
+
+private:
+    //!\brief The type of an iterator of the database sequence.
+    using database_iterator = iterator_t<database_type>;
+    //!\brief The alphabet type of the query sequence.
+    using query_alphabet_type = std::remove_reference_t<decltype(query[0])>;
+
+    //!\brief true - uses the ukkonen trick with the last active cell and bounds the error to config.max_errors.
+    static constexpr bool use_max_errors = detail::max_errors_concept<align_config_t>;
+    //!\brief Whether the alignment is a global alignment or not.
+    static constexpr bool is_global = detail::global_config_concept<align_config_t>;
+    //!\brief Whether the alignment is a semi-global alignment or not.
+    static constexpr bool is_semi_global = detail::semi_global_config_concept<align_config_t>;
+
+    //!\brief How to pre-initialize hp.
+    static constexpr word_type hp0 = is_global ? 1 : 0;
+
+    static_assert(8 * sizeof(word_type) <= 64, "we assume at most uint64_t as word_type");
+    static_assert((is_global && !is_semi_global) || (!is_global && is_semi_global), "Either set global or semi-global");
+
+    //!\brief The score of the alignment.
+    score_type _score{};
+    //!\brief The mask with a bit set at the position where the score change.
+    //!\details If #use_max_errors is true this corresponds to the last active cell.
+    word_type score_mask{0};
+    //!\brief The machine words which stores the positive vertical differences.
+    std::vector<word_type> vp{};
+    //!\brief The machine words which stores the negative vertical differences.
+    std::vector<word_type> vn{};
+    //!\brief The machine words which translate a letter of the query into a bit mask.
+    //!\details Each bit position which is true (=1) corresponds to a match of a letter in the query at this position.
+    std::vector<word_type> bit_masks{};
+
+    /*!\name Only used when use_max_errors is true
+     * \{
+     */
+    //!\brief Which score value is considered as a hit?
+    unsigned max_errors{255};
+    //!\brief The block containing the last active cell.
+    unsigned last_block{0};
+    //!\brief A mask with a bit set on the position of the last row.
+    word_type last_score_mask{};
+    //!\}
+
+    //!\brief The current position in the database.
+    database_iterator database_it{};
+    //!\brief The end position of the database.
+    database_iterator database_it_end{};
+
+    //!\brief The state of one column computation.
+    struct state_type
+    {
+        //!\copydoc pairwise_alignment_edit_distance_unbanded::vp
+        std::vector<word_type> vp{};
+        //!\copydoc pairwise_alignment_edit_distance_unbanded::vn
+        std::vector<word_type> vn{};
+    };
+    //!\brief The collection of each computation step.
+    std::vector<state_type> states{};
+    //!\brief Add a computation step
+    void add_state()
+    {
+        states.push_back(state_type{vp, vn});
+    }
+
+public:
+
+    /*!\name Constructors
```

Comment:
`Constructors, destructor and assignment`

Also please either default or delete the usual 5/6